wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -26,7 +26,7 @@ class ElasticsearchMapping(object):
         'DateField': 'date',
         'DateTimeField': 'date',
         'DecimalField': 'double',
-        'FileField': 'string',
+        'FileField': 'attachment',
```

Comment:
So there's now two ways we can index a ``FileField``. I believe at the moment we just index the filename.

We should probably keep the current behaviour as it is, and instead make the user explicitly define that they want the ``FileField`` indexed as an attachment by using the ``type``keyword argument on ``SearchField``.

For example:

``SearchField('myfile')`` will index the filename as a string (current behaviour)
``SearchField('myfile', type='attachment')`` will index the file contents using ``mapper-attachments``